### PR TITLE
Add link to run details page to export XAR

### PR DESCRIPTION
**Pull request by labkey-kevink**
- LABKEY.Experiment.exportRuns() API to download XAR of a set of runs
- Support downloading response of LABKEY.Ajax.request to a file

### CODE REVIEW

File: api/webapp/clientapi/core/Ajax.js [hunk 1]
```diff
@@ -26,6 +26,41 @@ LABKEY.Ajax = new function () {
...
 
+    var downloadFile = function (xhr, config) {
+        var filename = "";
+        if (typeof config.downloadFile === 'string') {
+            filename = config.downloadFile;
+        }
+        else {
+            var disposition = xhr.getResponseHeader('Content-Disposition');
+            if (disposition && disposition.indexOf('attachment') !== -1) {
+                var filenameRegex = /filename[^;=\n]*=((['"]).*?\2|[^;\n]*)/;
```
Comment (labkey-jeckels) [inline api/webapp/clientapi/core/Ajax.js:43]: Consider a comment on what this allows/disallows for those of us who aren't fluent in regex

File: experiment/src/org/labkey/experiment/controllers/exp/ExperimentController.java [hunk 2]
```diff
@@ -4397,46 +4407,24 @@ public void setDataIds(int[] dataIds)
...
-        }
 
         if (fileName == null || fileName.equals(""))
-        {
             fileName = "export.xar";
-        }
+
         fileName = fixupExportName(fileName);
         String xarXmlFileName = null;
         if (fileName.endsWith(".xar") || fileName.endsWith(".XAR") || fileName.endsWith("Xar"))
```
Comment (labkey-jeckels) [inline experiment/src/org/labkey/experiment/controllers/exp/ExperimentController.java]: I know this is preexisting, but what about making this fileName.toLowerCase().endWith()

**Review by labkey-jeckels — APPROVED** [review]
Couple of small suggestions. Didn't review the JS code in detail, assuming @labkey-nicka will handle that side.

File: experiment/src/org/labkey/experiment/controllers/exp/ExperimentController.java (hunk 2 above)
Comment (labkey-adam) [reply experiment/src/org/labkey/experiment/controllers/exp/ExperimentController.java]: Or ```StringUtils.endsWithIgnoreCase()```

**Pull request merged**